tests/unit/test_cv_device_tools.py

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/python
+# coding: utf-8 -*-
+# pylint: disable=logging-format-interpolation
+# pylint: disable=dangerous-default-value
+# pylint:disable=duplicate-code
+# flake8: noqa: W503
+# flake8: noqa: W1202
+# flake8: noqa: R0801
+
+from unittest.mock import call
+import pytest
+from tests.data.device_tools_unit import device_data, device_data_invalid
+from ansible_collections.arista.cvp.plugins.module_utils.device_tools import DeviceInventory, CvDeviceTools
+
+# list of paths to patch
+MOCK_LIST = [
+    'ansible_collections.arista.cvp.plugins.module_utils.device_tools.AnsibleModule',
+    'ansible_collections.arista.cvp.plugins.module_utils.device_tools.CvDeviceTools._CvDeviceTools__get_device']
+
+@pytest.fixture
+def setup(apply_mock, mock_cvpClient):
+    """
+    setup - setup method to apply mocks and patches
+    """
+    mock_ansible_module, mock__get_device = apply_mock(MOCK_LIST)
+    mock_ansible_module.fail_json.side_effect = mock_cvpClient.api.fail_json
+    cv_tools = CvDeviceTools(mock_cvpClient, mock_ansible_module)
+
+    return mock_ansible_module, mock__get_device, cv_tools
+
+@pytest.mark.state
+class TestDecommissionDevice():
+    """
+    Contains unit tests for decommission_device()
+    """
+
+    @pytest.mark.parametrize(
+        "device_data, expected, expected_fail_json_call_msg",
```

Comment:
The name expected sounds a bit weird - here and it is used in the test to take one branch or another which makes it difficult to understand what it is actually used for

Reply:
changed it to "flag"